src/states/node.rs

```diff
@@ -1457,10 +1457,10 @@ impl Node {
             }
         }
 
-        let peers_needing_tunnel = self.peer_mgr.peers_needing_tunnel();
-        if !peers_needing_tunnel.is_empty() &&
-           self.peer_mgr.potential_tunnel_nodes().contains(&(*public_id.name(), *peer_id)) {
-            for dst_id in peers_needing_tunnel {
+        for (dst_id, peer_name) in self.peer_mgr.peers_needing_tunnel() {
+            if self.peer_mgr
+                .potential_tunnel_nodes(&peer_name)
```

Comment:
but now we have to re-calculate for each peer now.  :(
the good thing is now we only looking for section, so hope it won't be too costly.